### PR TITLE
feat: add museum to networth calculation

### PR DESCRIPTION
**Pull request by DuckySoLucky**
## Description

Adds Museum to Networth Calculation

## Preview

> Before 
```
{
  noInventory: false,
  networth: 198724037488.5304,
  unsoulboundNetworth: 128993297609.74619,
  purse: 2217263443.4171467,
  bank: 25053770623.477383,
  types: {
    [...],
  }
}
```
> After
```
{
  noInventory: false,
  networth: 201220452628.549,
  unsoulboundNetworth: 129062097126.44618,
  purse: 2217263443.4171467,
  bank: 25053770623.477383,
  types: {
    [...],
    museum: {
      total: 2496415140.018611,
      unsoulboundTotal: 68799516.7,
      items: [Array]
    }
  }
}
```
> networth.types.museum.items: https://hastebin.com/share/gelonigaza.json

### CODE REVIEW

**Comment by MattTheCuber** [comment]
Thanks ducky!

File: helper/parseItems.js [hunk 1]
```diff
@@ -83,6 +83,27 @@ const postParseItems = async (profileData, items) => {
       items.pets.push(newPet);
     }
   }
+
+  // Parse Museum
+  items.museum = [];
+  if (museumData?.items) {
+    for (const data of Object.values(museumData.items)) {
+      if (data.borrowing === true) continue;
```
Comment (Altpapier) [inline helper/parseItems.js]: just remove the `=== true`

File: index.js [hunk 2]
```diff
@@ -7,14 +7,15 @@ const axios = require('axios');
  * Returns the networth of a profile
  * @param {object} profileData - The profile player data from the Hypixel API (profile.members[uuid])
  * @param {number} bankBalance - The player's bank balance from the Hypixel API (profile.banking?.balance)
+ * @param {object} museumData - The player's museum data from the Hypixel API (museum.members[uuid])
  * @param {{ cache: boolean, onlyNetworth: boolean, prices: object, returnItemData: boolean }} options - (Optional) cache: By default true (5 minute cache), if set to false it will always make a request to get the latest prices from github, onlyNetworth: If true, only the networth will be returned, prices: A prices object generated from the getPrices function. If not provided, the prices will be retrieved every time the function is called, returnItemData: If true, the item data will be returned in the object
  * @returns An object containing the player's networth calculation
  */
-const getNetworth = async (profileData, bankBalance, options) => {
+const getNetworth = async (profileData, bankBalance, museumData, options) => {
```
Comment (Altpapier) [inline index.js]: will break everyone who is updating and using options, some other options to include museumData idk. Also conflicts with the example

File: test/functions.js
Comment (Altpapier) [inline test/functions.js:1]: delete this

File: test/index.js
Comment (Altpapier) [inline test/index.js:1]: delete this

**Review by Altpapier — CHANGES_REQUESTED** [review]
is it also possible to not include museum data? (cba to test)

File: helper/parseItems.js [hunk 3]
```diff
@@ -83,6 +83,27 @@ const postParseItems = async (profileData, items) => {
...
+    for (const data of Object.values(museumData.items)) {
+      if (data.borrowing) continue;
+      if (data.items?.data === undefined) continue;
+
+      const decodedItem = await decodeData(data.items.data);
+
+      items.museum.push(...decodedItem);
+    }
+
+    for (const data of museumData.special) {
```
Comment (MattTheCuber) [inline helper/parseItems.js]: `museumData.special` can not exist

Fix:
`museumData.special` -> `museumData.special || []`

**Review by MattTheCuber — CHANGES_REQUESTED** [review]

File: helper/parseItems.js (hunk 3 above)
Comment (DuckySoLucky) [reply helper/parseItems.js]: fixed

File: examples/items.json
Comment (MattTheCuber) [inline examples/items.json:1]: So, `getPreDecodedNetworth` works with `museum` key in the `items` object? `postParseItems` makes it look like it wouldn't.

**Review by MattTheCuber — CHANGES_REQUESTED** [review]
Can you also update `examples/output.json`?

**Pull request merged**